src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -58,38 +58,53 @@ public BaseController(LookupDao inLookupDao, StreamingDao inStreamingDao, Naviga
 
 	protected void streamFlowLines(HttpServletResponse response,
 			String comid, String navigationMode, String stopComid, String distance, boolean legacy) throws Exception {
+
+		String featureType;
 		Map<String, Object> parameterMap = parameters.processParameters(comid, navigationMode, distance, stopComid);
-		FlowLineTransformer transformer = new FlowLineTransformer(response);
+
 		if (legacy) {
 			String sessionId = getSessionId(parameterMap, response);
 			if (null != sessionId) {
 				parameterMap.put(StreamingDao.SESSION_ID, sessionId);
-				addContentHeader(response);
-				streamResults(transformer, BaseDao.FLOW_LINES_LEGACY, parameterMap);
+				featureType = BaseDao.FLOW_LINES_LEGACY;
+			} else {
+				return;
 			}
 		} else {
-			addContentHeader(response);
-			streamResults(transformer, BaseDao.FLOW_LINES, parameterMap);
+			featureType = BaseDao.FLOW_LINES;
 		}
+
+		//Defer transformer creation to allow error messages to be sent (above) in the response if needed
+		FlowLineTransformer transformer = new FlowLineTransformer(response);
+
+		addContentHeader(response);
+		streamResults(transformer, featureType, parameterMap);
 	}
```

Comment:
This method and the next:
Defer creation of the Transformer until after logic and object creation.  Transformer creation calls getOutputStream() on the response, so would cause an exception in cases where an error might cause an error msg to be sent in the body.